addons/mail/static/src/messaging/component/notification_list/notification_list.js

```diff
@@ -93,6 +94,18 @@ class NotificationList extends Component {
                     uniqueId: thread.localId,
                 };
             });
+        if (props.filter === 'all') {
+            const notificationGroups = this.env.messaging.notificationGroupManager.groups;
+            notifications = Object.values(notificationGroups)
+                .sort((group1, group2) =>
+                    group1.date.isAfter(group2.date) ? -1 : 1
+                ).map(notificationGroup => {
+                    return {
+                        notificationGroup,
```

Comment:
I think it should observe `.__state`, and the getter `notifications` should no longer rely on `storeProps`.

Reply:
It should, but the problem is that thread preview is still relying on storeProps, and I don't want the current PR to change more than necessary in that regard.
This is basically task-2258605.